daisy/workflow/validate.go

```diff
@@ -141,6 +142,9 @@ func (w *Workflow) validateRequiredFields() error {
 	if w.Name == "" {
 		return errors.New("must provide workflow field 'Name'")
 	}
+	if !rfc1035Rgx.MatchString(strings.ToLower(w.Name)) {
```

Comment:
Not completely sure what is the best way for this. You decide. If it turns out to be too much of a hassle, we can talk more about it.